src/drt/src/pa/FlexPA_prep.cpp

```diff
@@ -2634,139 +2665,113 @@ bool FlexPA::genPatterns_commit(
-    frCoord left_pt = std::numeric_limits<frCoord>::max();
-    frCoord right_pt = std::numeric_limits<frCoord>::min();
-
-    const auto& [pin, inst_term] = pins[0];
-    const auto inst = inst_term->getInst();
-    for (auto& inst_term : inst->getInstTerms()) {
-      if (isSkipInstTerm(inst_term.get())) {
-        continue;
-      }
-      uint64_t n_no_ap_pins = 0;
-      for (auto& pin : inst_term->getTerm()->getPins()) {
-        if (pin_to_access_point.find(pin.get()) == pin_to_access_point.end()) {
-          n_no_ap_pins++;
-          pin_access_pattern->addAccessPoint(nullptr);
-        } else {
-          const auto& ap = pin_to_access_point[pin.get()];
-          const Point tmp_pt = ap->getPoint();
-          if (tmp_pt.x() < left_pt) {
-            left_access_point = ap;
-            left_pt = tmp_pt.x();
-          }
-          if (tmp_pt.x() > right_pt) {
-            right_access_point = ap;
-            right_pt = tmp_pt.x();
-          }
-          pin_access_pattern->addAccessPoint(ap);
+  std::set<frBlockObject*> owners;
+  if (target_obj != nullptr
+      && genPatterns_gc({target_obj}, objs, Commit, &owners)) {
+    pin_access_pattern->updateCost();
+    unique_inst_patterns_[curr_unique_inst_idx].push_back(
+        std::move(pin_access_pattern));
+    // genPatterns_print(nodes, pins, max_access_point_size);
+    is_valid = true;
+  } else {
+    for (int idx_1 = 0; idx_1 < (int) pins.size(); idx_1++) {
+      auto idx_2 = access_pattern[idx_1];
+      auto& [pin, inst_term] = pins[idx_1];
+      if (inst_term->hasNet()) {
+        if (owners.find(inst_term->getNet()) != owners.end()) {
+          viol_access_points.insert(std::make_pair(idx_1, idx_2));  // idx ;
         }
-      }
-      if (n_no_ap_pins == inst_term->getTerm()->getPins().size()) {
-        logger_->error(DRT, 91, "Pin does not have valid ap.");
-      }
-    }
-    pin_access_pattern->setBoundaryAP(true, left_access_point);
-    pin_access_pattern->setBoundaryAP(false, right_access_point);
-
-    std::set<frBlockObject*> owners;
-    if (target_obj != nullptr
-        && genPatterns_gc({target_obj}, objs, Commit, &owners)) {
-      pin_access_pattern->updateCost();
-      unique_inst_patterns_[curr_unique_inst_idx].push_back(
-          std::move(pin_access_pattern));
-      // genPatterns_print(nodes, pins, max_access_point_size);
-      is_valid = true;
-    } else {
-      for (int idx_1 = 0; idx_1 < (int) pins.size(); idx_1++) {
-        auto idx_2 = access_pattern[idx_1];
-        auto& [pin, inst_term] = pins[idx_1];
-        if (inst_term->hasNet()) {
-          if (owners.find(inst_term->getNet()) != owners.end()) {
-            viol_access_points.insert(std::make_pair(idx_1, idx_2));  // idx ;
-          }
-        } else {
-          if (owners.find(inst_term) != owners.end()) {
-            viol_access_points.insert(std::make_pair(idx_1, idx_2));  // idx ;
-          }
+      } else {
+        if (owners.find(inst_term) != owners.end()) {
```

Comment:
I wonder if there's a more efficient way to do this because this will traverse `owners` twice. It might not matter if the `else` is rarely exercised or if `owners` is small.